lib/rules/no-unnecessary-act.ts

```diff
@@ -52,17 +56,30 @@ export default createTestingLibraryRule<Options, MessageIds>({
     function getStatementIdentifier(statement: TSESTree.Statement) {
       const callExpression = getStatementCallExpression(statement);
 
-      if (!callExpression) {
+      if (
+        !callExpression &&
+        !isExpressionStatement(statement) &&
+        !isReturnStatement(statement)
+      ) {
         return null;
       }
 
-      const identifier = getDeepestIdentifierNode(callExpression);
+      if (callExpression) {
+        return getDeepestIdentifierNode(callExpression);
+      }
 
-      if (!identifier) {
-        return null;
+      if (
+        isExpressionStatement(statement) &&
+        isAwaitExpression(statement.expression)
+      ) {
+        return getPropertyIdentifierNode(statement.expression.argument);
+      }
+
+      if (isReturnStatement(statement) && statement.argument) {
```

Comment:
```suggestion
      if (isReturnStatement(statement) && Boolean(statement.argument)) {
```

Reply:
That doesn't work. We could do `!!statement.argument`, but what's the reasoning behind it?

![image](https://user-images.githubusercontent.com/34719980/126231965-708b99ff-42e4-4776-b08b-3088fc2c857e.png)

Reply:
I guess this is a good practice to force values being evaluated within an if statement to be purely boolean, but since we are not assigning them (just evaluating them within the if statement itself) I'd say it's ok to avoid casting to boolean.

When assigning values it could be dangerous `const foo = isReturnStatement(statement) && statement.argument` since `foo` could be a boolean or something else, but that's not a problem in the code we have.